File: app/src/main/java/com/sublime/dragplayer/model/Movie.java
package com.sublime.dragplayer.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by goonerdroid
 * on 22/12/17.
 */

public class Movie implements Parcelable{

    private String movieName;
    private int movieThumbnail;
    private String movieYear;
    private String movieGenre;


    public Movie(String movieName,int movieThumbnail,String movieYear,String movieGenre) {
        this.movieName = movieName;
        this.movieThumbnail = movieThumbnail;
        this.movieYear = movieYear;
        this.movieGenre = movieGenre;
    }

    private Movie(Parcel parcel) {
        movieName = parcel.readString();
        movieThumbnail = parcel.readInt();
        movieYear = parcel.readString();
        movieGenre = parcel.readString();
    }

    public static final Parcelable.Creator<Movie> CREATOR = new Parcelable.Creator<Movie>(){

        @Override
        public Movie createFromParcel(Parcel parcel) {
            return new Movie(parcel);
        }

        @Override
        public Movie[] newArray(int size) {
            return new Movie[0];
        }
    };


    public String getMovieName() {
        return movieName;
    }

    public int getMovieThumbnail() {
        return movieThumbnail;
    }

    public String getMovieYear() {
        return movieYear;
    }

    public String getMovieGenre() {
        return movieGenre;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(movieName);
        parcel.writeInt(movieThumbnail);
        parcel.writeString(movieYear);
        parcel.writeString(movieGenre);
    }
}

File: app/src/main/java/com/sublime/dragplayer/model/Movie.kt
package com.sublime.dragplayer.model

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by goonerdroid
 * on 16/1/18.
 */
class Movie(val movieName: String, val movieThumbnail: Int, val movieYear: String
            , val movieGenre: String, val movieTrailer: String) : Parcelable {


    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readInt(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(movieName)
        parcel.writeInt(movieThumbnail)
        parcel.writeString(movieYear)
        parcel.writeString(movieGenre)
        parcel.writeString(movieTrailer)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField val CREATOR = object : Parcelable.Creator<Movie>{
            override fun createFromParcel(parcel: Parcel): Movie {
                return Movie(parcel)
            }

            override fun newArray(size: Int): Array<Movie?> {
                return arrayOfNulls(size)
            }
        }
    }
}